### PR TITLE
Replacing NotNullAttribute with thrown exceptions

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNet.Hosting/WebHostBuilder.cs [hunk 1]
```diff
@@ -142,22 +165,32 @@ public WebHostBuilder UseServer(IServerFactory factory)
...
 
-        public WebHostBuilder UseStartup([NotNull] Type startupType)
+        public WebHostBuilder UseStartup(Type startupType)
         {
             if (startupType == null)
             {
                 throw new ArgumentNullException(nameof(startupType));
             }
+
+            if (startupType == null)
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/WebHostBuilder.cs]: duplicate

**Comment by Tratcher** [comment]
:shipit:

**Comment by tmds** [comment]
Is NotNullAttribute deprecated?

**Comment by pranavkm** [comment]
@tmds, the feature to support it isn't ready for v1.

**Comment by tmds** [comment]
I saw the announcement about 'dnu sources' and I thought it only had impact on the repository section on the json file. But it also means that Sources dependencies are not supported?

**Comment by pranavkm** [comment]
@tmds, no the feature that's not ready is discovery of precompiled modules in packages.Precompiled modules work when added as a source (which is how I'm generating these changes), but cannot be brought in from packages as yet.

**Comment by tmds** [comment]
ok, thanks

**Pull request closed without merging**